### PR TITLE
[WIP]  Add resourceGroupId to SessionConfigurationContext

### PR DESCRIPTION
**Pull request by wenleix**
The first two commits are from https://github.com/prestodb/presto/pull/8102

### CODE REVIEW

**Comment by wenleix** [comment]
@raghavsethi and @nezihyigitbasi : Would welcome some initial comments and thoughts :).

File: presto-main/src/main/java/com/facebook/presto/execution/QueryQueueManager.java [hunk 1]
```diff
@@ -26,5 +28,9 @@
 @ThreadSafe
 public interface QueryQueueManager
 {
-    void submit(Statement statement, QueryExecution queryExecution, Executor executor);
+    void submit(Statement statement, QueryExecution queryExecution, SelectionContext<?> group, Executor executor);
+
+    // TODO: Figure out what's the right thing to do for the following interface (e.g. keep it here? merge it into ResourceGroupManager
+    // and deprecate this interface?)
+    SelectionContext<?> selectGroup(SelectionCriteria criteria);
```
Comment (nezihyigitbasi) [inline presto-main/src/main/java/com/facebook/presto/execution/QueryQueueManager.java]: Once we remove the legacy queueing implementation I think there is no point in keeping this interface. We can have this public interface in `ResourceGroupManager`.

**Comment by nezihyigitbasi** [comment]
What's the motivation behind this change?

**Comment by wenleix** [comment]
Superseded by https://github.com/prestodb/presto/pull/10264

**Pull request closed without merging**